### PR TITLE
chore: add webcomponent draft

### PR DESCRIPTION
**Pull request by gexclaude**
This is a follow up pull request for https://github.com/asyncapi/asyncapi-react/pull/76 issued by @hesyar but sadly never merged.

It is about the same goal - providing a web component build making it possible to use the component outside of react applications. I just moved around the webpack config and npm build script.

### CODE REVIEW

File: library/src/containers/AsyncApi/AsyncApi.tsx [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import React, { Component } from 'react';
 import { Options as ParserOptions } from 'json-schema-ref-parser';
 
+// @ts-ignore
+import { register } from 'web-react-components';
```
Comment (magicmatatjahu) [inline library/src/containers/AsyncApi/AsyncApi.tsx:5]: Could we create separate file, maybe `AsyncApiWebComponent` and wrap `AsyncApiComponent` component from this file to new one? I mean making something like:

```js
import React, { Component } from 'react';
// @ts-ignore
import { register } from 'web-react-components';
import AsyncApiComponent from './AsyncApi';

export class AsyncApiWebComponent extends Component<AsyncApiProps> {
   constructor(props) {
      super(props);
    }

  render() {
    <>
        <style>@import '/assets/async-api/fiori.css';</style>
        <AsyncApiComponent {...this.props} />
    </>
  }
}

register(AsyncApiWebComponent, 'async-api-component', ['schema', 'config']);
```

Why is it better? If someone use React, then want only have dependencies related to react, not unnecessary deps like deps for web-component. Also final bundle for react app is bigger, and this is not good.

I also think that web-component should have separate package, but you can leave it to us :)

File: library/src/containers/AsyncApi/AsyncApi.tsx [hunk 2]
```diff
@@ -120,6 +123,7 @@ class AsyncApiComponent extends Component<AsyncApiProps, AsyncAPIState> {
         numberOfExpandedElement={initialExpandedElements}
       >
         <useChangeHashContext.Provider schemaName={bemClasses.getSchemaID()}>
+          <style>@import '/assets/async-api/fiori.css';</style>
```
Comment (magicmatatjahu) [inline library/src/containers/AsyncApi/AsyncApi.tsx:126]: Could we make it more generic (is it possible)? If someone wants use not fiori, but custom css file, so what then? :)

File: README.md [hunk 3]
```diff
@@ -66,6 +66,45 @@ import "@kyma-project/asyncapi-react/lib/styles/fiori.css";
...
+     {
+      "projects": {
+        "project-name": {
+          "architect": {
+            "build": {
+              "options": {
+                "scripts": [
+                  "node_modules/@webcomponents/webcomponentsjs/webcomponents-bundle.js",
+                  "src/assets/async-api/async-api-web-component.js"
+                ]
```
Comment (magicmatatjahu) [inline README.md:100]: Could you wrap this section to code block? ```

File: README.md [hunk 4]
```diff
@@ -66,6 +66,45 @@ import "@kyma-project/asyncapi-react/lib/styles/fiori.css";
...
+
+In an Angular component you can use the following way
+
+    <async-api-component [schema]="asyncApiDoc" [config]="asyncApiConfig"></async-api-component>
```
Comment (magicmatatjahu) [inline README.md:104]: Could you wrap this section to code block? ```

**Comment by magicmatatjahu** [comment]
@gexclaude Thanks for contribution! I left few comments. Could you refer to them?

**Comment by gexclaude** [comment]
> @gexclaude Thanks for contribution! I left few comments. Could you refer to them?

thank you, of course I'll go through them all.

**Comment by gexclaude** [comment]
@magicmatatjahu I updated this PR, thank you for having a look.

**Comment by gexclaude** [comment]
@magicmatatjahu did you already got the chance to take a look at the updated changes?

**Comment by magicmatatjahu** [comment]
@gexclaude I'm sorry, but when I found the free time in last week, I checked if the component was working. It took me a while because I wanted to check if it will work with Angular :) Sorry again, because I could write that it will take me some time.

Some thoughts about PR:
- it works with Angular, so very good, but `@webcomponents/webcomponentsjs/webcomponents-bundle.js` is unnecessary, because for Angular you can use `CUSTOM_ELEMENTS_SCHEMA` from `@angular/core`, but it should stay for es5 compatibility and other libs/frameworks, but we have to add to the documentation that dev should first download this package (I have an error that package didn't exist)
- the documentation should state that styles should go to the `src/asyncapi/fiori.css` folder if someone doesn't upload/use their own styles.

But overall very good and thank you very much for this PR :)

First I would like to create infra for new package and then publish one. If that's not a problem for you, can I merge this PR, make some changes and then publish?

**Comment by gexclaude** [comment]
@magicmatatjahu 

no worries, I just want to make sure we proceed.

yes, it isn't perfect yet regarding what you mentioned. resulting web component is also quite big, but for us it is a good first start.

of course, I would be more than happy if we can get it merged.

**Comment by gexclaude** [comment]
@magicmatatjahu anything I can help with before merge of this PR?

**Comment by gexclaude** [comment]
@magicmatatjahu ping - anything I can help to get this PR merged? :-)

File: README.md [hunk 5]
```diff
@@ -66,6 +66,55 @@ import "@kyma-project/asyncapi-react/lib/styles/fiori.css";
...
+      "projects": {
+        "project-name": {
+          "architect": {
+            "build": {
+              "options": {
+                "scripts": [
+                  "node_modules/@webcomponents/webcomponentsjs/webcomponents-bundle.js",
+                  "src/assets/async-api/async-api-web-component.js"
+                ]
+ ```
```
Comment (magicmatatjahu) [inline README.md:108]:
```suggestion
```
```

**Comment by magicmatatjahu** [comment]
@gexclaude I was on vacation and sorry for my previous promise. I didn't have much time. It will merge and I'll start preparing infra for pushing the component. Sorry again.

**Comment by magicmatatjahu** [comment]
PR has problems with dependencies, but it will be fixed in next commit.

**Comment by gexclaude** [comment]
> @gexclaude I was on vacation and sorry for my previous promise. I didn't have much time. It will merge and I'll start preparing infra for pushing the component. Sorry again.

Many thanks, @magicmatatjahu - no need to be sorry, I just wanted to proceed with a slightly "push" - if I knew you were on vacation I would have been more patient though.

**Comment by magicmatatjahu** [comment]
@all-contributors please add @gexclaude for code, platform, ideas

**Comment by magicmatatjahu** [comment]
@all-contributors please add @derberg for code, infra, bug, blog, maintenance

**Comment by magicmatatjahu** [comment]
@all-contributors please add @magicmatatjahu for infra, bug, example, maintenance

**Pull request merged**